featuretools/primitives/standard/rolling_transform_primitive.py

```diff
@@ -0,0 +1,343 @@
+import numpy as np
+import pandas as pd
+from woodwork.column_schema import ColumnSchema
+from woodwork.logical_types import Datetime, Double
+
+from featuretools.primitives.base.transform_primitive_base import (
+    TransformPrimitive
+)
+from featuretools.primitives.utils import _roll_series_with_gap
+from featuretools.utils.gen_utils import Library
+
+
+class RollingMax(TransformPrimitive):
+    """Determines the maximum of entries over a given window.
+
+    Description:
+        Given a list of numbers and a corresponding list of
+        datetimes, return a rolling maximum of the numeric values,
+        starting at the row `gap` rows away from the current row and looking backward
+        over the specified window (by `window_length` and `gap`).
+
+        Input datetimes should be monotonic.
+
+    Args:
+        window_length (int): The number of rows to be included in each frame. For data
+            with a uniform sampling frequency, for example of one day, the window_length will
+            correspond to a period of time, in this case, 7 days for a window_length of 7.
+        gap (int, optional): The number of rows prior to each instance to be skipped before
+            beginning the window over which the maximum is determined. Defaults to 0, which
+            will include each instance in the window.
+        min_periods (int, optional): Minimum number of observations required for a window to have a value.
+            Can only be as large as window_length. Defaults to 1.
+
+    Examples:
+        >>> import pandas as pd
+        >>> rolling_max = RollingMax(window_length=3)
+        >>> times = pd.date_range(start='2019-01-01', freq='1min', periods=5)
+        >>> rolling_max(times, [4, 3, 2, 1, 0]).tolist()
+        [4.0, 4.0, 4.0, 3.0, 2.0]
+
+        We can also control the gap before the rolling calculation.
+
+        >>> import pandas as pd
+        >>> rolling_max = RollingMax(window_length=3, gap=1)
+        >>> times = pd.date_range(start='2019-01-01', freq='1min', periods=5)
+        >>> rolling_max(times, [4, 3, 2, 1, 0]).tolist()
+        [NaN, 4.0, 4.0, 3.0, 2.0]
+
+        We can also control the minimum number of periods required for the rolling calculation.
+
+        >>> import pandas as pd
+        >>> rolling_max = RollingMax(window_length=3, min_periods=3)
+        >>> times = pd.date_range(start='2019-01-01', freq='1min', periods=5)
+        >>> rolling_max(times, [4, 3, 2, 1, 0]).tolist()
+        [NaN, NaN, 4.0, 3.0, 2.0]
+    """
+    name = "rolling_max"
+    input_types = [ColumnSchema(logical_type=Datetime, semantic_tags={'time_index'}), ColumnSchema(semantic_tags={'numeric'})]
+    return_type = ColumnSchema(logical_type=Double, semantic_tags={'numeric'})
+    compatibility = [Library.PANDAS, Library.DASK, Library.KOALAS]
+
+    def __init__(self, window_length=3, gap=0, min_periods=1):
```

Comment:
Previously the default window-length was `1d`, which makes sense and will be a valid window with multiple instances present if the data's sampling frequency is less than 1 day. At this point in the implementation, though, we do not have access to date offsets, and setting the window length to `1` doesn't make much sense, as it would never end up with more than one instance present in the window. 

I think I might prefer using `window_length=7` or something that has some time-related meaning. But chose 3 for the moment just to show the arbitrary nature of having to set a default for this parameter. 

Also, when we implement #1757, we can go back to setting the default to be `1d`, but for now, I don't think there's a way to ensure this won't be a breaking change when that's implemented.